tools/doc/preprocess.js

```diff
@@ -50,9 +50,10 @@ function processIncludes(inputFile, input, cb) {
 
         // Add comments to let the HTML generator know how the anchors for
         // headings should look like.
-        includeData[fname] = `<!-- [start-include:${fname}] -->\n` +
-                             inc + `\n<!-- [end-include:${fname}] -->\n`;
-        input = input.split(include + '\n').join(includeData[fname] + '\n');
+        includeData[fname] = `<!-- [start-include:${fname}] -->
```

Comment:
Multi-line string concatenation is actually ok (the way it was before), but it would be nice if the 2nd line became this instead:

```js
`${inc}\n<!-- [end-include:${fname}] -->\n`
```

That way we don't have to lose indentation level.